docs/en/ingest-management/fleet/api-generated/template/index.mustache

```diff
@@ -0,0 +1,170 @@
+////
+This content is generated from the open API specification.
+Any modifications made to this file will be overwritten.
+////
+
+++++
+<div class="openapi">
+  <h2>Access</h2>
+  {{#hasAuthMethods}}
+    <ol>
+    {{#authMethods}}
+      <li>{{#isBasic}}HTTP Basic Authentication{{/isBasic}}{{#isOAuth}}OAuth AuthorizationUrl:{{authorizationUrl}}TokenUrl:{{tokenUrl}}{{/isOAuth}}{{#isApiKey}}APIKey KeyParamName:{{keyParamName}} KeyInQuery:{{isKeyInQuery}} KeyInHeader:{{isKeyInHeader}}{{/isApiKey}}</li>
+    {{/authMethods}}
+    </ol>
+  {{/hasAuthMethods}}
+
+  <h2><a name="__Methods">Methods</a></h2>
+  [ Jump to <a href="#__Models">Models</a> ]
+
+  {{! for the tables of content, I cheat and don't use CSS styles.... }}
+  <h3>Table of Contents </h3>
+  <div class="method-summary">{{access}}</div>
+  {{#apiInfo}}
+  {{#apis}}
+  {{#operations}}
+  <h4><a href="#{{baseName}}">{{baseName}}</a></h4>
+  <ul>
+  {{#operation}}
+  <li><a href="#{{nickname}}"><code><span class="http-method">{{httpMethod}}</span> {{path}}</code></a></li>
+  {{/operation}}
+  </ul>
+  {{/operations}}
+  {{/apis}}
+  {{/apiInfo}}
+
+  {{#apiInfo}}
+  {{#apis}}
+  {{#operations}}
```

Comment:
Is there a way Fleet can control the order in which operations are rendered? It'd be great to order them the same way they appear in the spec file as we group endpoints by "namespace" - e.g. all routes prefixed with `/epm` are colocated, `/setup` appears first, all `/agent_policy` routes are colocated, etc. 

![image](https://user-images.githubusercontent.com/6766512/221651248-3d030c8d-f86f-40d1-8729-41391b70482a.png)

Here's the order in the spec, for reference: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/common/openapi/entrypoint.yaml#L15-L153. Making this a predictably ordered list that aligns with the spec file would be awesome, so happy to do anything we can on Fleet's side to make that work.

Reply:
Happily, I have discovered the output order is controlled by the tags in `bundled.json`. For example, I added `"0001"` to `/setup`, `"0002"` to `/settings` and and so on for the first ten or so endpoints (if that's the right term). Like this:

```
  "paths": {
    "/setup": {
      "post": {
        "summary": "Setup",
        "tags": ["0001"],
        "responses": {
          "200": {
```

The output now renders like this:
![Screenshot 2023-02-27 at 4 01 32 PM](https://user-images.githubusercontent.com/41695641/221686676-bedc14fa-3ca2-4e94-82cb-de081c220ac5.png)

A couple of things I noticed:

1. The `post /health_check` endpoint appears in the Fleet API docs if I give it a tag, and if I remove that tag, it disappears. Since I don't see that endpoint in the Swagger docs, I assume it's meant to be hidden. So, I guess we need to be careful about which endpoints have or don't have tags, assuming at least some aren't intended to be exposed in the docs.

2. Some endpoints, such as "/agents/action_status" don't have a tag field at all, so perhaps one can be added.

@kpollich If you don't mind adjusting the spec to add tags I'd be very happy to retest.

Reply:
Thanks for digging into this @kilfoyle - I think we can add some tags to group everything. Realistically, I don't care much about alphabetical order or anything as much as I care about grouping related endpoints together, which should be easily achievable by doing things like `tags: ['agent-policies']` for each `/agent_policies/*` endpoint and so forth.

> The post /health_check endpoint appears in the Fleet API docs if I give it a tag, and if I remove that tag, it disappears. Since I don't see that endpoint in the Swagger docs, I assume it's meant to be hidden.

Regarding `/health_check` - we probably don't want this documented in the public facing API spec, you're right. I'll probably just remove its path value from the spec.

I'll get a PR together with these changes and provide you a new spec file shortly. Many thanks!

Reply:
@kilfoyle - Here's that PR: https://github.com/elastic/kibana/pull/152386 - please find the updated `bundled.yaml` or `bundled.json` attached to that PR and feel free to make a pass here with the updated spec. I took some time to clean up tags, descriptions, etc to hopefully group things consistently. 

Here's an example with the official Swagger UI tool: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/elastic/kibana/27b14796ddb26a390614396bfac1ba9190511786/x-pack/plugins/fleet/common/openapi/bundled.yaml#/. I'm sure we can go further in organizing here, but hopefully this represents a solid first step forward 🙂

Reply:
@kpollich - Thanks for the terrific changes! I think the docs are [looking great](https://observability-docs_2715.docs-preview.app.elstc.co/guide/en/fleet/master/fleet-apis.html). I added the headers back in and they display the groups quite nicely.

I tested bundled.json and bundled.yaml. Both work nicely with exactly the same results.

One thing remaining would be to hide the "FleetInternals" section, but otherwise I think we're in pretty good shape now.